utils/hash.ts

```diff
@@ -0,0 +1,13 @@
+export function truncateHash(hash: string, parts: number = 4, chars: number = 4): string {
+  const blockLength = (hash.length - chars) / (parts - 1)
+  const result = []
+
+  while(result.length < parts - 1) {
+    const startPosition = result.length * blockLength
+    result.push(hash.slice(startPosition, startPosition + chars))
+  }
+
+  result.push(hash.slice(hash.length - chars, hash.length))
+  
+  return result.join('...')
+}
```

Comment:
Not a blocker, but on `website-testnet` I had need to have the full hash (for copying purposes) and also render it nicely -- you may want to approach things similarly here? https://github.com/iron-fish/website-testnet/blob/master/components/user/EventRow.module.css#L51-L56